tests/unit_tests/agents/test_agent.py

```diff
@@ -268,6 +268,44 @@ def test_agent_tool_return_direct_in_intermediate_steps() -> None:
     assert action.tool == "Search"
 
 
+def test_agent_treat_exception() -> None:
+    responses = [
+        "My action is to throw an exception",
+    ]
+    fake_llm = FakeListLLM(responses=responses)
+    agent = initialize_agent(
+        load_tools(["exception"]),
+        fake_llm,
+        agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION,
+        verbose=True,
+        max_iterations=1,
+    )
+    output = agent.run("when was langchain made")
+    assert "Agent stopped" in output
+
+    # now test it does throw exception without the tool
+    tools = [
+        Tool(
+            name="Search",
+            func=lambda x: x,
+            description="Useful for searching",
+        )
+    ]
+    fake_llm = FakeListLLM(responses=responses)
+    agent = initialize_agent(
+        tools,
+        fake_llm,
+        agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION,
+        verbose=True,
+        max_iterations=1,
+    )
+    try:
```

Comment:
```suggestion
with pytest.raises(OutputParserException)
 ...
 ```
 
 or ValueError if we decide not to make a custom error